sources/sections/05-date-time-format.adoc

```diff
@@ -8,14 +8,16 @@ Internet protocols.
 === Ordering
 
 If date and time components are ordered from least precise to most
-precise, then a useful property is achieved. Assuming that the time
-zones of the dates and times are the same (e.g., all in UTC),
+precise, then a useful property is achieved. Assuming that the UTC
+offsets of the dates and times are the same (e.g., all in UTC) and
 expressed using the same string (e.g., all "Z" or all "+00:00"), all
-times have the same number of fractional second digits, and they all
-have the same suffix (or none), then the date and time strings may be
+times have the same number of fractional second digits, all time
+designators use the same string (e.g., "T"), and all years
+are in the same era and use the same number of digits, then they may be
```

Comment:
With this do you mean that they are either all using the extended years format or the 4-digit year format?

Reply:
"In the same era" means the years must all be nonnegative or all be negative—that lexicographic sorting of a collection containing a _mixture_ of years on or after 0000 with years before it is not guaranteed to result in a chronological sort (e.g., "**+002021-03-08T13:31:37Z**" < "**+002021-03-09T02:00:00Z**" < "**-000043-03-15T11:23:42Z**" < "**-000322-06-11T03:14:15Z**").

"Use the same number of digits" means the years must all be non-expanded (4-digit) or all be expanded with a sign and the same number of additional digits—that lexicographic sorting of a collection containing a _mixture_ of values in which the year is not always expressed using the same number of digits is not guaranteed to result in a chronological sort (e.g., "**+000275760-09-13T00:00:00Z**" < "**+002021-03-08T13:31:37Z**" < "**2021-03-09T02:00:00Z**").

But if all years _are_ in the same era (before 0000 vs. on or after it) and are represented using the same number of digits (four vs. six or possibly more), then lexicographic sorting corresponds with chronological sorting as long as the other constraints also hold.